src/context.js

```diff
@@ -0,0 +1,153 @@
+import * as objects from './ponyfills/objects.js';
+import { emit } from './waitforit.js';
+
+export const CONTEXT_CHANGED = 'context.changed';
+export const CONTEXT_INITIALIZED = 'context.initialized';
+export const CONTEXT_VALUE_REMOVED = 'context.value.removed';
+export const CONTEXT_VALUE_ADDED = 'context.value.added';
+export const CONTEXT_VALUE_CHANGED = 'context.value.changed';
+export const CONTEXT_DESTROYED = 'context.destroyed';
+
+/**
+ * The EvolvContext provides functionality to manage data relating to the client state, or context in which the
+ * variants will be applied.
+ *
+ * This data is used for determining which variables are active, and for general analytics.
+ *
+ * @constructor
+ */
+function EvolvContext() {
```

Comment:
No need for action necessarily, just a critique: I think I understand why you might have done it, but if this and similar functions were meant to be treated as constructors, I think they should have been written as classes (in the ES5 manner, with `Object.create()`, etc.). Doing so breaks assumptions that consuming libraries might make.